shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -32,7 +33,7 @@ function dropTracking3pCookiesFromResponse (request) {
     const tab = tabManager.get({ tabId: request.tabId })
     let responseHeaders = request.responseHeaders
 
-    if (tab && request.type !== 'main_frame') {
+    if (startup.readySync() && tab && request.type !== 'main_frame') {
```

Comment:
Because [they recommend not to do stuff like that for MV3](https://developer.chrome.com/docs/extensions/mv3/migrating_to_service_workers/#event_listeners), at the risk that your event listener doesn't get registered. Don't ask me how that all works in the browser itself though!

Reply:
ah ok. MV3 goodness... A couple of follow ups:

1. This means we have a brief blindspot whenever the service worker restarts, right? Particularly if we're commonly getting restarted by this event (as it's attached to a webRequest listener, this should trigger constantly).
2. Can this specific listener even work in MV3? This is cookie blocking code, which requires blocking webRequests.

Reply:
I was thinking about this again last night and you're quite right:
- The `readyBool` variable should really be stored in session storage, otherwise we are [replacing one MV3 incompatible pattern with another](https://developer.chrome.com/docs/extensions/mv3/migrating_to_service_workers/#state).
- I'd rather not wait on session storage needlessly in these event listeners. Hopefully reading from session storage is fast, but these requests are blocking on our block/allow decision.
- These event listeners are MV2 only, like you say.

I'm generally trying to remove MV3 incompatible patterns from the code, even when they're not important, so I'd obviously rather not add another one. But it seems like we're damned either way here. Perhaps the best option would be to wait for `readySync` before creating the event listeners, but to leave a comment explaining that this pattern is MV2 only and should not be copied for code that needs to run on MV3. What do you think?

Reply:
OK I have gone with the simpler approach and limited the changes to the third party cookie code. I considered also waiting for `startup.ready()` before registering the main `handleRequest()` listener, but that seems risky given that the event is used to maintain the extension's tab state too.